src/shared/generator.c

```
@@ -370,54 +370,61 @@ int generator_write_fsck_deps(
+                log_warning_errno(r, "%s: failed to parse timeout, ignoring: %s", what, timeout);
                 return 0;
         }
 
         node = fstab_node_to_udev_node(what);
         if (!node)
                 return log_oom();
         if (!is_device_path(node)) {
-                log_warning("x-systemd.device-timeout ignored for %s", what);
+                log_warning("x-systemd.device-timeout ignored for %s.", what);
                 return 0;
         }
 
         r = unit_name_from_path(node, ".device", &unit);
-        if (r < 0)
-                return log_error_errno(r, "Failed to make unit name from path: %m");
+        if (r < 0) {
+                log_error_errno(r, "%s: failed to make unit name from path '%s', ignoring: %m", what, node);
+                return 0;
+        }
 
-        return write_drop_in_format(dir, unit, 50, "device-timeout",
-                                    "# Automatically generated by %s\n"
-                                    "# from supplied options \"%s\"\n\n"
-                                    "[Unit]\n"
-                                    "JobRunningTimeoutSec=%s",
-                                    program_invocation_short_name,
-                                    opts,
-                                    timeout);
+        r = write_drop_in_format(dir, unit, 50, "device-timeout",
+                                 "# Automatically generated by %s\n"
+                                 "# from supplied options \"%s\"\n\n"
+                                 "[Unit]\n"
+                                 "JobRunningTimeoutSec=%s",
+                                 program_invocation_short_name,
+                                 opts,
+                                 timeout);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0)
+                log_warning_errno(r, "%s: failed to write drop-in with device timeout, ignoring: %m", what);
```

Hmm, let's just ignore all errors here.

---

I don't think it matters much either way, but OK.